omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -176,41 +176,32 @@ where
                     }
                 }
             };
-            self.state = (Role::Follower, Phase::Accept);
-            self.current_seq_num = accsync.seq_num;
-            let cached_idx = self.outgoing.len();
-            self.latest_accepted_meta = Some((accsync.n, cached_idx));
-            self.outgoing.push(PaxosMessage {
-                from: self.pid,
-                to: from,
-                msg: PaxosMsg::Accepted(accepted),
-            });
             match accsync.stopsign {
                 Some(ss) => {
                     if let Some(ss_entry) = self
                         .internal_storage
                         .get_stopsign()
                         .expect("storage error while trying to read stopsign")
                     {
-                        let StopSignEntry {
-                            decided: has_decided,
-                            stopsign: _my_ss,
-                        } = ss_entry;
-                        if !has_decided {
+                        if !ss_entry.decided(old_decided_idx) {
                             self.accept_stopsign(ss);
                         }
                     } else {
                         self.accept_stopsign(ss);
                     }
-                    let a = AcceptedStopSign { n: accsync.n };
-                    self.outgoing.push(PaxosMessage {
-                        from: self.pid,
-                        to: from,
-                        msg: PaxosMsg::AcceptedStopSign(a),
-                    });
+                    accepted.accepted_idx += 1;
```

Comment:
Here I would like to add an `assert_eq!(ss.accepted_index, accepted.accepted_idx +1)` (it's ok to add an assert here since we only handle acceptsync when leader changes). And then set `accepted.accepted_idx = ss.accepted_index`